### PR TITLE
Docs Ingestion page refactor

### PR DESCRIPTION
**Pull request by techdocsmith**
Refactor ingestion into smaller topics. Clean up some style here and there.

### Description

This PR breaks down the ingestion landing page topic into smaller topics. Some benefits include:
- Ingestion concepts like the Data Model, Rollup, and Partitioning now have smaller more focused topics.
- The Ingestion spec reference stands alone as a reference topic.
- Should make it easier to add conceptual or example material for a specific concept.

There may be some trade off in topic maintenance. For example if there's an enhancement to rollup, we may need to edit `rollup.md` and `ingestion-spec.md`.

This PR has:
- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by techdocsmith** [comment]
@sthetland & @vtlim , PTAL

File: docs/ingestion/index.md [hunk 1]
```diff
@@ -88,656 +79,5 @@ This table compares the three available options:
...
-|metricCompression|Compression format for primitive type metric columns. Options are `lz4`, `lzf`, `uncompressed`, or `none` (which is more efficient than `uncompressed`, but not supported by older versions of Druid).|`lz4`|
-|longEncoding|Encoding format for long-typed columns. Applies regardless of whether they are dimensions or metrics. Options are `auto` or `longs`. `auto` encodes the values using offset or lookup table depending on column cardinality, and store them with variable size. `longs` stores the value as-is with 8 bytes each.|`longs`|
-
-Beyond these properties, each ingestion method has its own specific tuning properties. See the documentation for each
-[ingestion method](#ingestion-methods) for details.
+| **[Rollup modes](rollup.md)** | Perfect if `forceGuaranteedRollup` = true in the [`tuningConfig`](native-batch.md#tuningconfig).  | Always perfect. | Perfect if `forceGuaranteedRollup` = true in the [`tuningConfig`](native-batch.md#tuningconfig). |
+| **Partitioning options** | Dynamic, hash-based, and range-based partitioning methods are available. See [Partitions Spec](./native-batch.md#partitionsspec) for details.|
```
Comment (sthetland) [inline docs/ingestion/index.md:83]: This row is missing the two columns on the right. I think this would fix it, but pls double-check me and test the table after committing:
 
```suggestion
| **Partitioning options** | Dynamic, hash-based, and range-based partitioning methods are available. See [Partitions Spec](./native-batch.md#partitionsspec) for details.| Hash-based or range-based partitioning via [`partitionsSpec`](hadoop.md#partitionsspec). | Dynamic and hash-based partitioning methods are available. See [Partitions Spec](./native-batch.md#partitionsspec-1) for details. |
```

**Review by sthetland — CHANGES_REQUESTED** [review]
Good move splitting up the pages. Comments below.

File: docs/ingestion/rollup.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
+id: rollup
+title: "Data rollup"
+sidebar_label: Data rollup
+description: Introducdes rollup as a concept. Provides suggestions to maximize the benefits of rollup. Differentiates between perfect and best-effort rollup.
+---
+Druid can roll up data ingestion time to reduce the amount of raw data to  store on disk. Rollup is a form of summarization or pre-aggregation. Rolling up data your can dramatically reduce the size of data to be store and reducing row counts by potentially orders of magnitude. The trade off for the efficiency of rollup means you lose the ability to query individual events.
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/rollup.md:7]: Much better I think.

File: docs/ingestion/index.md (hunk 1 above)
Comment (techdocsmith) [reply docs/ingestion/index.md]: Looks good in `npm run start`

File: docs/ingestion/rollup.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+---
+id: rollup
+title: "Data rollup"
+sidebar_label: Data rollup
+description: Introduces rollup as a concept. Provides suggestions to maximize the benefits of rollup. Differentiates between perfect and best-effort rollup.
+---
+Druid can roll up data at ingestion time to reduce the amount of raw data to  store on disk. Rollup is a form of summarization or pre-aggregation. Rolling up data can dramatically reduce the size of data to be stored and reduce row counts by potentially orders of magnitude. As a trade off for the efficiency of rollup, you lose the ability to query individual events.
```
Comment (vtlim) [inline docs/ingestion/rollup.md:7]:
```suggestion
Druid can roll up data at ingestion time to reduce the amount of raw data to  store on disk. Rollup is a form of summarization or pre-aggregation. Rolling up data can dramatically reduce the size of data to be stored and reduce row counts by potentially orders of magnitude. As a trade-off for the efficiency of rollup, you lose the ability to query individual events.
```

File: docs/ingestion/rollup.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+In this query, `cnt` refers to a "count" type metric from your ingestion spec. See
+[Counting the number of ingested events](schema-design.md#counting) on the "Schema design" page for more details about how counting works when rollup is enabled.
+
+Tips for maximizing rollup:
+
+- Design your schema with fewer dimensions and lower cardinality dimensions to yield better rollup ratios.
+- Use [sketches](schema-design.md#sketches) to avoid storing high cardinality dimensions, which decrease rollup ratios.
+- Adjust your `queryGranularity` at ingestion time to increase the chances that multiple rows in Druid having matching timestamps. For example, use five minute query granularity (`PT5M`) instead of one minute (`PT1M`).
+- You can optionally load the same data into more than one Druid datasource. For example:
+    - Create a "full" datasource that has rollup disabled, or enabled, but with a minimal rollup ratio
```
Comment (vtlim) [inline docs/ingestion/rollup.md:30]:
```suggestion
    - Create a "full" datasource that has rollup disabled, or enabled, but with a minimal rollup ratio.
```

File: docs/ingestion/rollup.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     When queries only involve dimensions in the "abbreviated" set, use the second datasource to reduce query times. Often, this method only requires a small increase in storage footprint because abbreviated datasources tend to be substantially smaller.
+- If you use a [best-effort rollup](#perfect-rollup-vs-best-effort-rollup) ingestion configuration that does not guarantee perfect rollup, try one of the following:
+    - Switch to a guaranteed perfect rollup option.
+    - [Reindex](data-management.md#reingesting-data) or [compact](compaction.md) your data in the background after initial ingestion.
+
+## Perfect rollup vs Best-effort rollup
+
+Depending on the ingestion method, Druid has the following rollup options:
+- Guaranteed _perfect rollup_: Druid perfectly aggregates input data at ingestion time.
+- _best-effort rollup_: Druid may not perfectly aggregate input data. Therefore, multiple segments might contain rows with the same timestamp and dimension values.
```
Comment (vtlim) [inline docs/ingestion/rollup.md:41]:
```suggestion
## Perfect rollup vs best-effort rollup

Depending on the ingestion method, Druid has the following rollup options:
- Guaranteed _perfect rollup_: Druid perfectly aggregates input data at ingestion time.
- _Best-effort rollup_: Druid may not perfectly aggregate input data. Therefore, multiple segments might contain rows with the same timestamp and dimension values.
```

File: docs/ingestion/rollup.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+In general, ingestion methods that offer best-effort rollup do this for one of the following reasons:
+- The ingestion method parallelizes ingestion without a shuffling step required for perfect rollup.
+- The ingestion method uses _incremental publishing_ which means it finalizes and publishes segments before all data for a time chunk has been received,
```
Comment (vtlim) [inline docs/ingestion/rollup.md:45]:
```suggestion
- The ingestion method uses _incremental publishing_ which means it finalizes and publishes segments before all data for a time chunk has been received.
```

File: docs/ingestion/partitioning.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+id: partitioning
+title: Partitioning
+sidebar_label: Partitioning
+description: Describes time chunk and secondary partitioning in Druid. Provides guidance to choose a secondary partition dimension.
+---
+
+Optimal partitioning and sorting of segments within your Druid datasources can have substantial impact on footprint and performance.
```
Comment (vtlim) [inline docs/ingestion/partitioning.md:8]: phrasing seems a bit strange; another option could be "partitioning of and sorting segments" though not much better

File: docs/ingestion/partitioning.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+One way to partition is to your load data into separate datasources. This is a perfectly viable approach that works very well when the number of datasources does not lead to excessive per-datasource overheads. 
+
+This topic describes how to set up partitions within a single datasource. It does not cover using multiple datasources. See [Multitenancy considerations](../querying/multitenancy.md) for more details on splitting data into separate datasources and potential operational considerations.
```
Comment (vtlim) [inline docs/ingestion/partitioning.md:12]:
```suggestion
This topic describes how to set up partitions within a single datasource. It does not cover how to use multiple datasources. See [Multitenancy considerations](../querying/multitenancy.md) for more details on splitting data into separate datasources and potential operational considerations.
```

File: docs/ingestion/partitioning.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+dimension that you often use as a filter when possible. Such partitioning often improves compression and query performance. For example, some cases have yielded threefold storage size decreases.
+
+## Partitioning and sorting
+
+Partitioning and sorting work well together. If you do have a "natural" partitioning dimension, consider placing it first in the `dimensions` list of your `dimensionsSpec`. This way Druid sorts rows within each segment by that column. This sorting configuration frequently improves compression more than using partitioning alone.
+
+> Note that Druid always sorts rows within a segment by timestamp first, even before the first dimension listed in your `dimensionsSpec`. This sorting can preclude the efficacy of dimension sorting. To work around this limitation if necessary, set your `queryGranularity` equal to `segmentGranularity` in your [`granularitySpec`](./ingestion-spec.md#granularityspec). Druid will set all timestamps within the segment to the same value, and letting you identify a [secondary timestamp](schema-design.md#secondary-timestamps) as the "real" timestamp.
```
Comment (vtlim) [inline docs/ingestion/partitioning.md:29]:
```suggestion
> Note that Druid always sorts rows within a segment by timestamp first, even before the first dimension listed in your `dimensionsSpec`. This sorting can preclude the efficacy of dimension sorting. To work around this limitation if necessary, set your `queryGranularity` equal to `segmentGranularity` in your [`granularitySpec`](./ingestion-spec.md#granularityspec). Druid will set all timestamps within the segment to the same value, letting you identify a [secondary timestamp](schema-design.md#secondary-timestamps) as the "real" timestamp.
```

File: docs/ingestion/partitioning.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+## How to configure partitioning
+
+Not all ingestion methods support an explicit partitioning configuration, and not all have equivalent levels of flexibility. If you are doing initial ingestion through a less-flexible method like
+Kafka), you can use [reindexing](data-management.md#reingesting-data) or [compaction](compaction.md) to repartition your data after initial ingestion. This is a powerful technique you can use to optimally partition any data older than a certain even while you continuously add new data from a stream.
```
Comment (vtlim) [inline docs/ingestion/partitioning.md:34]: "older than a certain **event**" or "older than a certain **X**, even while" ?

File: docs/ingestion/partitioning.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+The following table shows how each ingestion method handles partitioning:
+
+|Method|How it works|
+|------|------------|
+|[Native batch](native-batch.md)|Configured using [`partitionsSpec`](native-batch.md#partitionsspec) inside the `tuningConfig`.|
+|[Hadoop](hadoop.md)|Configured using [`partitionsSpec`](hadoop.md#partitionsspec) inside the `tuningConfig`.|
+|[Kafka indexing service](../development/extensions-core/kafka-ingestion.md)|Kafka topic partitioning defines how partitions the datasource. You can also [reindex](data-management.md#reingesting-data) or [compact](compaction.md) to repartition after initial ingestion.|
+|[Kinesis indexing service](../development/extensions-core/kinesis-ingestion.md)|Kinesis stream sharding defines how partitions the datasource.. You can also [reindex](data-management.md#reingesting-data) or [compact](compaction.md) to repartition after initial ingestion.|
```
Comment (vtlim) [inline docs/ingestion/partitioning.md:43]: "defines how partitions the datasource" sounds unclear

File: docs/querying/multi-value-dimensions.md [hunk 12]
```diff
@@ -1,4 +1,4 @@
----
+  ---
```
Comment (vtlim) [inline docs/querying/multi-value-dimensions.md:1]:
```suggestion
---
```

**Review by vtlim — COMMENTED** [review]
I noticed a few other things I couldn't edit, will make the changes on my end and push to your branch

File: docs/ingestion/data-model.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+Druid stores data in datasources, which are similar to tables in a traditional relational database management systems (RDBMS). Druid's data model shares  similarities with both relational and timeseries data models.
+
+## Primary timestamp
+
+Druid schemas must always include a primary timestamp. Druid uses the primary timestamp to [partition and sort](./partitioning.md) your data. Druid uses the primary timestamp to rapidly identify and retrieve data within the time range of queries. Druid also uses the primary timestamp column
+for time-based [data management operations](./data-management.md) such as dropping time chunks, overwriting time chunks, and time-based retention rules.
+
+Druid parses the primary timestamp based on the [`timestampSpec`](./ingestion-spec.md#timestampspec) configuration at ingestion time. You can control other important operations that are based on the primary timestamp
+[`granularitySpec`](./ingestion-spec.md#granularityspec). Regardless of the source input field for the primary timestamp, Druid always stores the timestamp in the `__time` column in your Druid datasource.
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/data-model.md:16]: No. `granlularitySpec` controls other aspects not controlled by the `timestampspec`. I think that this will be better to move the last sentence starting with "Regardless to line 15 after ingestion time. Then start a new paragraph for `granularitySpec`. I'll make this change after you push yours.

File: docs/ingestion/partitioning.md (hunk 11 above)
Comment (techdocsmith) [reply docs/ingestion/partitioning.md:43]:
```suggestion
|[Kafka indexing service](../development/extensions-core/kafka-ingestion.md)|Kafka topic partitioning defines how Druid partitions the datasource. You can also [reindex](data-management.md#reingesting-data) or [compact](compaction.md) to repartition after initial ingestion.|
|[Kinesis indexing service](../development/extensions-core/kinesis-ingestion.md)|Kinesis stream sharding defines how Druid partitions the datasource. You can also [reindex](data-management.md#reingesting-data) or [compact](compaction.md) to repartition after initial ingestion.|
```

Missing `Druid`. @vtlim

File: docs/ingestion/partitioning.md (hunk 10 above)
Comment (techdocsmith) [reply docs/ingestion/partitioning.md:34]:
```suggestion
Kafka), you can use [reindexing](data-management.md#reingesting-data) or [compaction](compaction.md) to repartition your data after initial ingestion. This is a powerful technique you can use to optimally partition any data older than a certain time threshold while you continuously add new data from a stream.
```

File: docs/ingestion/data-model.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+Druid stores data in datasources, which are similar to tables in a traditional relational database management system (RDBMS). Druid's data model shares  similarities with both relational and timeseries data models.
+
+## Primary timestamp
+
+Druid schemas must always include a primary timestamp. Druid uses the primary timestamp to [partition and sort](./partitioning.md) your data. Druid uses the primary timestamp to rapidly identify and retrieve data within the time range of queries. Druid also uses the primary timestamp column
+for time-based [data management operations](./data-management.md) such as dropping time chunks, overwriting time chunks, and time-based retention rules.
+
+Druid parses the primary timestamp based on the [`timestampSpec`](./ingestion-spec.md#timestampspec) configuration at ingestion time. Regardless of the source field field for the primary timestamp, Druid always stores the timestamp in the `__time` column in your Druid datasource.
+
+You can control other important operations that are based on the primary timestamp
```
Comment (techdocsmith) [inline docs/ingestion/data-model.md:17]:
```suggestion
You can control other important operations that are based on the primary timestamp in the
```

File: docs/ingestion/data-model.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+Druid stores data in datasources, which are similar to tables in a traditional relational database management system (RDBMS). Druid's data model shares  similarities with both relational and timeseries data models.
+
+## Primary timestamp
+
+Druid schemas must always include a primary timestamp. Druid uses the primary timestamp to [partition and sort](./partitioning.md) your data. Druid uses the primary timestamp to rapidly identify and retrieve data within the time range of queries. Druid also uses the primary timestamp column
+for time-based [data management operations](./data-management.md) such as dropping time chunks, overwriting time chunks, and time-based retention rules.
+
+Druid parses the primary timestamp based on the [`timestampSpec`](./ingestion-spec.md#timestampspec) configuration at ingestion time. Regardless of the source field field for the primary timestamp, Druid always stores the timestamp in the `__time` column in your Druid datasource.
```
Comment (vtlim) [inline docs/ingestion/data-model.md:15]:
```suggestion
Druid parses the primary timestamp based on the [`timestampSpec`](./ingestion-spec.md#timestampspec) configuration at ingestion time. Regardless of the source field for the primary timestamp, Druid always stores the timestamp in the `__time` column in your Druid datasource.
```

File: docs/ingestion/index.md [hunk 16]
```diff
@@ -22,33 +22,24 @@ title: "Ingestion"
...
-(or the [Indexer](../design/indexer.md) processes) load your source data. One exception is
-Hadoop-based ingestion, where this work is instead done using a Hadoop MapReduce job on YARN (although MiddleManager or Indexer
-processes are still involved in starting and monitoring the Hadoop jobs). 
+For most ingestion methods, the Druid [MiddleManager](../design/middlemanager.md) processes or the [Indexer](../design/indexer.md) processes load your source data. One exception is
+Hadoop-based ingestion, which uses a Hadoop MapReduce job on YARN MiddleManager or Indexer processes to start and monitor Hadoop jobs. 
 
-Once segments have been generated and stored in [deep storage](../dependencies/deep-storage.md), they are loaded by Historical processes. 
-For more details on how this works, see the [Storage design](../design/architecture.md#storage-design) section 
-of Druid's design documentation.
+After Druid creates segments and stores them in [deep storage](../dependencies/deep-storage.md), Historical processes load them to respond to queries. See the [Storage design](../design/architecture.md#storage-design) section of the Druid design documentation for more information.
```
Comment (loquisgon) [inline docs/ingestion/index.md:30]: We already said in first paragraph that Druid creates segments. Now we are are saying that they get stored in a special phase. I would rephrase this as 

"Segments created by the ingestion process get stored in [deep storage...] which in turn are loaded in Historical nodes by Historical processes in order to respond to Historical queries" 

At some point the distinction has to be made between queries served by historical processes and those served by real time (i.e. middle manager/indexer) processes. BTW the latter only happens for streaming ingestion.

File: docs/ingestion/index.md [hunk 17]
```diff
@@ -22,33 +22,24 @@ title: "Ingestion"
...
 
-This **page you are currently reading** provides information about universal Druid ingestion concepts, and about
-configurations that are common to all [ingestion methods](#ingestion-methods).
-
-The **individual pages for each ingestion method** provide additional information about concepts and configurations
-that are unique to each ingestion method.
-
-We recommend reading (or at least skimming) this universal page first, and then referring to the page for the
-ingestion method or methods that you have chosen.
+For additional information about concepts and configurations that are unique to each ingestion method, see the topic for the ingestion method.
```
Comment (loquisgon) [inline docs/ingestion/index.md:38]: Insert a paragraph: 

"This topic also describe the two main ways of ingest data in Druid: streaming and batch"...blah blah blah

File: docs/ingestion/index.md [hunk 18]
```diff
@@ -22,33 +22,24 @@ title: "Ingestion"
...
 
 ## Ingestion methods
 
-The table below lists Druid's most common data ingestion methods, along with comparisons to help you choose
+The tables below list Druid's most common data ingestion methods, along with comparisons to help you choose
```
Comment (loquisgon) [inline docs/ingestion/index.md:42]: Below in the table, we introduce "Suervisor type" with no explanation about what a supervisor is...can we link that to where we explain what a supervisor is?

File: docs/ingestion/index.md [hunk 19]
```diff
@@ -88,656 +79,5 @@ This table compares the three available options:
...
-|dimensionCompression|Compression format for dimension columns. Options are `lz4`, `lzf`, or `uncompressed`.|`lz4`|
-|metricCompression|Compression format for primitive type metric columns. Options are `lz4`, `lzf`, `uncompressed`, or `none` (which is more efficient than `uncompressed`, but not supported by older versions of Druid).|`lz4`|
-|longEncoding|Encoding format for long-typed columns. Applies regardless of whether they are dimensions or metrics. Options are `auto` or `longs`. `auto` encodes the values using offset or lookup table depending on column cardinality, and store them with variable size. `longs` stores the value as-is with 8 bytes each.|`longs`|
-
-Beyond these properties, each ingestion method has its own specific tuning properties. See the documentation for each
-[ingestion method](#ingestion-methods) for details.
+| **[Rollup modes](rollup.md)** | Perfect if `forceGuaranteedRollup` = true in the [`tuningConfig`](native-batch.md#tuningconfig).  | Always perfect. | Perfect if `forceGuaranteedRollup` = true in the [`tuningConfig`](native-batch.md#tuningconfig). |
```
Comment (loquisgon) [inline docs/ingestion/index.md:82]: I wonder if it is better to say here: "perfect or best effort depending on batch ingestion method & other setting -- see rollup for more info"

File: docs/ingestion/index.md [hunk 20]
```diff
@@ -88,656 +79,5 @@ This table compares the three available options:
...
+| **Partitioning options** | Dynamic, hash-based, and range-based partitioning methods are available. See [Partitions Spec](./native-batch.md#partitionsspec) for details.| Hash-based or range-based partitioning via [`partitionsSpec`](hadoop.md#partitionsspec). | Dynamic and hash-based partitioning methods are available. See [Partitions Spec](./native-batch.md#partitionsspec-1) for details. |
```
Comment (loquisgon) [inline docs/ingestion/index.md:83]: Use consistent naming for anchors: partitionSpec vs "Partition Spec"

File: docs/ingestion/data-formats.md [hunk 21]
```diff
@@ -128,6 +128,8 @@ The CSV `inputFormat` has the following components:
 
 ### TSV (Delimited)
 
+The `inputFormat` to load data of a delimited format. An example is:
```
Comment (loquisgon) [inline docs/ingestion/data-formats.md:131]: An example for `inputForm` using JSON data is:

Comment (loquisgon) [reply docs/ingestion/data-formats.md:131]: Also this phrase "The `inputFormat` to load data of a delimited format. An example is:" keeps being repeated after each type is introduced...it feel repetitive and unecessary

File: docs/ingestion/data-formats.md [hunk 22]
```diff
@@ -521,8 +523,8 @@ An example `flattenSpec` is:
 }
 ```
 > Conceptually, after input data records are read, the `flattenSpec` is applied first before
```
Comment (loquisgon) [inline docs/ingestion/data-formats.md:525]: I did not see a section that explains "flattenSpec"

File: docs/ingestion/data-model.md [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+dimensions like a set of columns to ingest. The dimensions behave exactly as you would expect from any database that does not support a rollup feature.
+
+At ingestion time, you configure dimensions in the [`dimensionsSpec`](./ingestion-spec.md#dimensionsspec).
+
+## Metrics
+
+Metrics are columns that Druid stores in an aggregated form. Metrics are most useful when you enable [rollup](rollup.md). If you specify a metric, you can apply an aggregation function to each row during ingestion. This
+has the following benefits:
+
+- With [rollup](rollup.md) enabled, Druid collapses multiple rows into one row even while retaining summary information. For example, the [rollup tutorial](../tutorials/tutorial-rollup.md) demonstrates using rollup to collapse netflow data to a single row per `(minute, srcIP, dstIP)` tuple, while retaining aggregate information about total packet and byte counts.
```
Comment (loquisgon) [inline docs/ingestion/data-model.md:35]: "even" here does not seem appropriate. Maybe this: "Rollup is a form of aggregation that collapses dimensions while aggregating the values in the metrics, that is, it collapses rows but retains its summary information."

**Comment by loquisgon** [comment]
All of my comments are content related so I delegate to your criteria if you want to address them or not. I am fine with the separation of sections. so LGTM.

File: docs/ingestion/data-formats.md (hunk 22 above)
Comment (loquisgon) [reply docs/ingestion/data-formats.md:525]: I found it after a while: [`flattenSpec`](../../ingestion/ingestion-spec.md#flattenspec) maybe insert an anchor to it?

Comment (loquisgon) [reply docs/ingestion/data-formats.md:525]: I found it after a while: [`flattenSpec`](../../ingestion/ingestion-spec.md#flattenspec) maybe insert an anchor to it?

File: docs/ingestion/index.md (hunk 18 above)
Comment (techdocsmith) [reply docs/ingestion/index.md:42]: There is nowhere in the docs that describes a supervisor 😞 . I added my own brief definition.

File: docs/ingestion/index.md (hunk 20 above)
Comment (techdocsmith) [reply docs/ingestion/index.md:83]: Normally I'd prefer the English "Partitions spec" to the code, but the header is `partitionsSpec` so went with that.

File: docs/ingestion/data-formats.md (hunk 21 above)
Comment (techdocsmith) [reply docs/ingestion/data-formats.md:131]: Ok. This was fun 😬 Cleaned it up for the most part, PTAL

File: docs/ingestion/data-formats.md (hunk 22 above)
Comment (techdocsmith) [reply docs/ingestion/data-formats.md:525]: This is the section on flattenSpec and it goes in the inputFormat object. I couldn't find a better place to link.

File: docs/querying/query-context.md [hunk 24]
```diff
@@ -61,6 +61,7 @@ Unless otherwise noted, the following parameters apply to all query types.
...
 |enableJoinLeftTableScanDirect|`false`|This flag applies to queries which have joins. For joins, where left child is a simple scan with a filter,  by default, druid will run the scan as a query and the join the results to the right child on broker. Setting this flag to true overrides that behavior and druid will attempt to push the join to data servers instead. Please note that the flag could be applicable to queries even if there is no explicit join. since queries can internally translated into a join by the SQL planner.|
+|debug| `false` | Flag indicating whether to enable debugging outputs for the query. When set to false, no additional logs will be produced (logs produced will be entirely dependent on your logging level). When set to true, the following addition logs will be produced:<br />- Log the stack trace of the exception (if any) produced by the query |
```
Comment (sthetland) [inline docs/querying/query-context.md:64]:
```suggestion
|debug| `false` | Flag indicating whether to enable debugging outputs for the query. When set to false, no additional logs will be produced. (Logs produced will be entirely dependent on your logging level.) When set to true, the following additional logs will be produced:<br />- Log the stack trace of the exception (if any) produced by the query |
```

File: docs/development/extensions-core/protobuf.md [hunk 25]
```diff
@@ -112,82 +112,86 @@ Important supervisor properties
...
+                "useFieldDiscovery": true
+            },
+            "binaryAsString": false
+        },
+        "taskCount": 1,
+        "replicas": 1,
+        "taskDuration": "PT1H",
+        "type": "kafka"
+    }
+}
```
Comment (sthetland) [inline docs/development/extensions-core/protobuf.md:194]: An extra `}` may have gotten added here.

**Review by sthetland — CHANGES_REQUESTED** [review]
Again, nice refactor. A few nits from some spot checking, but also I think a merge conflict snuck through.

File: docs/development/extensions-core/protobuf.md (hunk 25 above)
Comment (techdocsmith) [reply docs/development/extensions-core/protobuf.md:194]: I didn't touch these so no idea how this happened. Not sure why these last 3 even wound up in this PR

Comment (techdocsmith) [reply docs/development/extensions-core/protobuf.md:194]: This passes a JSON parser, so I'm not going to change it b/c it doesn't relate to this PR.

File: docs/ingestion/data-formats.md [hunk 26]
```diff
@@ -215,21 +218,22 @@ The `inputFormat` to load data of Parquet format. An example is:
...
+For more informtion on how Druid handles Avro types, see [Avro Types](../development/extensions-core/avro.md#avro-types) section for
 
-> You need to include the [`druid-avro-extensions`](../development/extensions-core/avro.md) as an extension to use the Avro Stream input format.
+Configure the Avro `inputFormat` to load Avro data as follows:
 
-> See the [Avro Types](../development/extensions-core/avro.md#avro-types) section for how Avro types are handled in Druid
+| Field | Type | Description | Required |
+|-------|------|-------------|----------|
+|type| String| This should be set to `avro_stream` to read Avro serialized data| yes |
+|flattenSpec| JSON Object |Define a [`flattenSpec`](#flattenspec) to extract nested values from a Avro record. Note that only 'path' expression are supported ('jq' is unavailable).| no (default will auto-discover 'root' level properties) |
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/data-formats.md:232]: I didn't touch this. Can we fix it in another PR>

File: docs/ingestion/data-formats.md [hunk 27]
```diff
@@ -422,11 +427,20 @@ Multiple Instances:
...
-> See the [Avro Types](../development/extensions-core/avro.md#avro-types) section for how Avro types are handled in Druid
+See the [Avro Types](../development/extensions-core/avro.md#avro-types) section for how Avro types are handled in Druid
 
-The `inputFormat` to load data of Avro OCF format. An example is:
+Configure the Avro OCF `inputFormat` to load Avro OCF data as follows:
+
+| Field | Type | Description | Required |
+|-------|------|-------------|----------|
+|type| String| This should be set to `avro_ocf` to read Avro OCF file| yes |
+|flattenSpec| JSON Object |Define a [`flattenSpec`](#flattenspec) to extract nested values from a Avro records. Note that only 'path' expression are supported ('jq' is unavailable).| no (default will auto-discover 'root' level properties) |
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/data-formats.md:439]: I didn't touch this one, can we fix in another PR

File: docs/querying/multi-value-dimensions.md [hunk 28]
```diff
@@ -22,9 +22,21 @@ title: "Multi-value dimensions"
   ~ under the License.
   -->
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/querying/multi-value-dimensions.md]: OK I missed this

**Comment by techdocsmith** [comment]
This PR got out of sync beyond repair. #11576 carries it. cc: @suneet-s , @sthetland
